app/javascript/components/chat/Conversation.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+
+import ConversationItem from './ConversationItem';
+
+class Conversation extends Component {
+  componentDidMount() {
+    this.scrollToBottom();
+  }
+  
+  componentDidUpdate() {
+    this.scrollToBottom();
+  }
+
+  renderTypingMessage = () => {
+    const typers = _.map(this.props.typers, 'username')
+
+    let result;
+
+    switch (typers.length) {
+      case 0:
+        result = ''
+        break;
+      case 1:
+        result = `${typers[0]} is typing ...`
+        break;
+      case 2:
+          result = `${typers.join(' and ')} are typing ...`
+          break;
+      default:
+        const othersCount = typers.length - 1
+        result = `${typers[0]} and ${othersCount} other people are typing ...`
+    }
+
+    return <span>{result}</span>;
+  }
+
+  scrollToBottom = () => {
+    this.messagesEnd.scrollIntoView();
+  }
+
+  render() {
+    const { messages, currentUserId } = this.props;
+
+    return (
+      <div className="conversation">
+        <div className="conversation__content">
+          {messages.length > 0 ? messages.map(message => (
```

Comment:
new line after '?' like in https://reactjs.org/docs/conditional-rendering.html#inline-if-else-with-conditional-operator